modules/shared_data/src/test/scala/org/elpaca_metagraph/shared_data/daemons/fetcher/XFetcherSuite.scala

```diff
@@ -0,0 +1,107 @@
+package org.elpaca_metagraph.shared_data.daemons.fetcher
+
+import cats.effect._
+import cats.effect.testkit.TestControl
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import io.circe.syntax._
+import org.elpaca_metagraph.shared_data.Utils
+import org.elpaca_metagraph.shared_data.app.ApplicationConfig
+import org.elpaca_metagraph.shared_data.types.Lattice.{LatticeUser, XAccount}
+import org.elpaca_metagraph.shared_data.types.States.XDataSource
+import org.elpaca_metagraph.shared_data.types.X._
+import org.http4s._
+import org.http4s.circe._
+import org.http4s.client.Client
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.epoch.EpochProgress
+
+import java.time.LocalDateTime
+import scala.collection.immutable.ListMap
+
+class XFetcherSuite extends AnyFunSuite with Matchers with FetcherSuite {
```

Comment:
Both terms should reward only once and they should be marked as rewarded to the first term in the post, so if another post has both tags again, it will be rewarded for the second term.